### PR TITLE
Integrate RON

### PR DESCRIPTION
**Pull request by torkleyy**

### CODE REVIEW

**Comment by Xaeroxe** [comment]
looks good to me, but for some reason the documentation is failing to generate?

File: src/gfx_device/display_config.rs [hunk 1]
```diff
@@ -1,25 +1,31 @@
...
+    #[serde(default)]
+    pub max_dimensions: Option<(u32, u32)>,
+    /// Enables or disables vertical synchronization.
+    #[serde(default)]
+    pub vsync: bool,
+    /// Level of MSAA anti-aliasing.
+    #[serde(default)]
+    pub multisampling: u16,
+    /// Sets the visibility of the window.
+    #[serde(default = "true")]
```
Comment (kvark) [inline src/gfx_device/display_config.rs]: I think it doesn't like this `"true"`. Perhaps, quoting should not be here?

**Comment by Xaeroxe** [comment]
@torkleyy https://serde.rs/attr-default.html Woops, it doesn't seem to like specifying literals like that.

File: src/gfx_device/display_config.rs (hunk 1 above)
Comment (torkleyy) [inline src/gfx_device/display_config.rs]: Yeah, I know that it doesn't work yet. For whatever reason it accepts a function name. I'll try to make a PR for `#[serde(default_value=...)]`.

**Comment by Xaeroxe** [comment]
@torkleyy While we wait for the `default_value` suggestion to land, how do you feel about the commit I made just now?

**Comment by Xaeroxe** [comment]
And now we have another problem :/

**Comment by Xaeroxe** [comment]
Ah.  serde_derive doesn't work without serde.

**Comment by vitvakatu** [comment]
@Xaeroxe stable travis passed, neat

**Comment by torkleyy** [comment]
Yeah, we can use it for now.

**Comment by torkleyy** [comment]
But examples won't work right now.

**Comment by Xaeroxe** [comment]
@torkleyy mind if I fix them?

**Comment by Xaeroxe** [comment]
Uh woops, sorry that was supposed to go to my branch, I'll take it off

**Pull request merged**